core/src/Cuda/Kokkos_Cuda_Parallel.hpp

```diff
@@ -303,7 +303,9 @@ class TeamPolicyInternal<Kokkos::Cuda, Properties...>
     return m_thread_scratch_size[level];
   }
 
-  inline typename traits::execution_space space() const { return m_space; }
+  inline const typename traits::execution_space& space() const {
```

Comment:
Why is this one inlined?

Reply:
I would prefer to call the member function `execution_space()` (I don't have a strong opinion on this)

Reply:
some execution policies already call it `space()` and define it as inline, so I believe this is just being consistent?  see: https://github.com/kokkos/kokkos/blob/develop/core/src/Kokkos_ExecPolicy.hpp#L113-L115

Reply:
> some execution policies already call it space()

Fair enough.

> and define it as inline, so I believe this is just being consistent?

The `inline` is missing for `TeamPolicy::space()` and in all other specializations of `TeamPolicyInternal`.  I would drop the inline specifier because I find it superfluous for a function defined in the class body.

Reply:
There are a lot more member functions declared as `inline` in the class body here. I am happy to remove just this one in this pull request and maybe tackle the others separately.